Importer.pm

```diff
@@ -247,36 +247,24 @@ sub needsUpdate { if (!main::SCANNER) {
 	my ($class, $cb) = @_;
 
 	my $lastScanTime = $cache->get('tidal_library_last_scan') || return $cb->(1);
-
-	my $checkFav = sub {
-		my ($userId, $type, $previous, $acb) = @_;
-
-		Plugins::TIDAL::API::Async->new({
-			userId => $userId
-		})->getLatestCollectionTimestamp(sub {
-			my $timestamp = shift;
-			$acb->($timestamp > $lastScanTime);
-		}, $type);
-	};
-
-	my $workers = [ map {
-		my $userId = $_;
-		my @tasks = (
-			sub { $checkFav->($userId, 'albums', @_) },
-			sub { $checkFav->($userId, 'artists', @_) },
-		);
-
-		if (!$class->ignorePlaylists) {
-			push @tasks, sub { $checkFav->($userId, 'playlists', @_) };
-		}
-
-		@tasks;
-	} sort {
-		$a <=> $b
-	} values %{_enabledAccounts()} ];
-
-	Async::Util::achain(
-		steps => $workers,
+	my @keys = qw(updatedFavoriteArtists updatedFavoriteTracks updatedFavoriteAlbums);
+	push @keys, qw(updatedFavoritePlaylists updatedPlaylists) unless $class->ignorePlaylists;
+
+	Async::Util::amap(
```

Comment:
It's just that my understanding we did not want to do them all at the same time for all users. The difference is that we don't need to have many workers for each type (album, artist ...) because we know with one call if any of these has been changed for one user. I just did one at a time to gently check every user, but it could be more than that or even a single sync call because at the end it might not be much work. Just do a for each user that will create callbacks for each so I don't think we even need amap. My 2 cents

Reply:
fixed